<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitReportReporter -->
<testsuite hostname="MrB" failures="0" tests="2" name="roughTrials.JDBCMysqlTest" time="1.475" errors="1" timestamp="2021-03-10T23:24:59 EST" skipped="0">
  <testcase classname="roughTrials.JDBCMysqlTest" name="mysqlSelect" time="0.125"/>
  <system-out/>
  <testcase classname="roughTrials.JDBCMysqlTest" name="mysqlDML" time="1.350">
    <error message="Cannot add or update a child row: a foreign key constraint fails (`traffic_violations_db`.`officers`, CONSTRAINT `FK_Officers` FOREIGN KEY (`OfficerID`) REFERENCES `violations` (`OfficerID`))" type="java.sql.SQLIntegrityConstraintViolationException">
      <![CDATA[java.sql.SQLIntegrityConstraintViolationException: Cannot add or update a child row: a foreign key constraint fails (`traffic_violations_db`.`officers`, CONSTRAINT `FK_Officers` FOREIGN KEY (`OfficerID`) REFERENCES `violations` (`OfficerID`))
at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:117)
at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97)
at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
at com.mysql.cj.jdbc.StatementImpl.executeInternal(StatementImpl.java:764)
at com.mysql.cj.jdbc.StatementImpl.execute(StatementImpl.java:648)
at roughTrials.JDBCMysqlTest.mysqlDML(JDBCMysqlTest.java:29)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:564)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:599)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:174)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:822)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:147)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
at java.base/java.lang.Thread.run(Thread.java:832)
]]>
    </error>
  </testcase> <!-- mysqlDML -->
  <system-out/>
</testsuite> <!-- roughTrials.JDBCMysqlTest -->
